File: src/Python/575.DistributeCandies.py

class Solution:
    def distributeCandies(self, candies):
        """
        :type candies: List[int]
        :rtype: int
        """
        return min(len(set(candies)), len(candies) // 2)

if __name__ == '__main__':
    solution = Solution()
    print(solution.distributeCandies([1,1,2,2,3,3]))
    print(solution.distributeCandies([1,1,2,3]))
else:
    pass

File: src/CSharp/575.DistributeCandies.cs

using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.DistributeCandies
{
    public class Solution
    {
        public int DistributeCandies(int[] candies)
        {
            int[] hash = new int[200001];
            int hashsize = 0;
            foreach (var c in candies)
            {
                if (hash[c + 100000] == 0)
                {
                    hash[c + 100000] = 1;
                    hashsize++;
                }
            }
            return Math.Min(hashsize, candies.Length / 2);
        }
    }
}

